### PR TITLE
Fix linalg compilation, update compiler and fix rng for stock

### PR DESCRIPTION
**Pull request by DenisScherbakov**
Compilation error:
linalg.cc(105): error: namespace "std" has no member "vector"
      std::vector<std::string> benches;

### CODE REVIEW

**Review by antonwolfy — APPROVED** [review]

**Review by antonwolfy — DISMISSED** [review]

File: numpy/random/Makefile [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 BENCHMARKS = rng
 SOURCES = $(addsuffix .c,$(BENCHMARKS))
-CC = icc
+CC = icx
 CLANG_FORMAT = clang-format
-CFLAGS += -m64 -fPIC -fomit-frame-pointer -xSSE4.2 -axCORE-AVX2,CORE-AVX512 \
+CFLAGS += -m64 -fPIC -fomit-frame-pointer -xSSE4.2 -axCORE-AVX2,COMMON-AVX512 \
```
Comment (oleksandr-pavlyk) [inline numpy/random/Makefile]: Maybe we should leave `CORE-AVX512`.

File: numpy/random/rng.py [hunk 2]
```diff
@@ -46,31 +41,56 @@ def sample_hypergeom(rs, sz):
...
 OUTER_REPS=6
 INNER_REPS=512
+SEED=123
 
 def main():
     import itertools
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--prefix',  required=False, default="IntelPython",     help="Print with each result")
+    parser.add_argument('--impl',  required=False, default="mkl", choices=['mkl', 'numpy'], help='RNG implementation\n'
```
Comment (oleksandr-pavlyk) [inline numpy/random/rng.py]:
```suggestion
    parser.add_argument('--impl',  required=False, default="mkl_random", choices=['mkl_random', 'numpy'], help='RNG implementation\n'
```

File: numpy/linalg/Makefile [hunk 3]
```diff
@@ -2,9 +2,8 @@
 #
 # SPDX-License-Identifier: MIT
 
-CXX = icc
-CXXFLAGS = -O3 -g -xCORE-AVX2 -axCOMMON-AVX512 \
-	   -qopt-report=5 -qopt-report-phase=openmp,par,vec
+CXX = icpx
+CXXFLAGS = -O3 -g -xSSE4.2 -axCORE-AVX2,COMMON-AVX512 -qopt-report
```
Comment (oleksandr-pavlyk) [inline numpy/linalg/Makefile]:
```suggestion
CXXFLAGS = -O3 -g -xSSE4.2 -axCORE-AVX2,CORE-AVX512 -qopt-report
```

File: numpy/umath/Makefile [hunk 4]
```diff
@@ -2,9 +2,10 @@
 #
 # SPDX-License-Identifier: MIT
 
-CC = icc
-CFLAGS = -qopenmp -xCORE-AVX2 -axCOMMON-AVX512 -O3 \
-	 -g -lmkl_rt -Wall -pedantic
+CC = icx
+CFLAGS = -qopenmp -xSSE4.2 -axCORE-AVX2,COMMON-AVX512 -O3 \
```
Comment (oleksandr-pavlyk) [inline numpy/umath/Makefile]:
```suggestion
CFLAGS = -qopenmp -xSSE4.2 -axCORE-AVX2,CORE-AVX512 -O3 \
```

**Review by oleksandr-pavlyk — COMMENTED** [review]
Once changes are addressed it is good to merge

**Review by oleksandr-pavlyk — APPROVED** [review]

**Pull request merged**